plugins/default/lib/component_health.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python
+import sys
+import json
+import argparse
+import urllib2
+
+import nagios
+
+class RequestError(Exception):
+    def __init__(self, message):
+        self.message = message
+
+def generate_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-H","--host", action="store", required=True, help="Host")
+    parser.add_argument('-P','--port', action="store", required=True, help="Port")
+    parser.add_argument("-E","--endpoint", action="store", required=False, help="Endpoint", default="/sys/info/health")
+    return parser
+
+def do_request(host, port, endpoint):
+    url =  'http://%s:%s%s' % (host,port,endpoint)
+    try:
+        data = urllib2.urlopen(url).read()
+    except urllib2.HTTPError, error:
+        if error.code == 500:
+            # For some reason the health endpoint returns a http status code of 500 when the health isn't ok, so we have to check the http error for a health response here
+            try:
+                data = error.read()
+                json.loads(data)
+            except ValueError:
+                raise RequestError(error)
+        else:
+            raise RequestError(error)
+    except urllib2.URLError:
+        raise RequestError('The health endpoint at "%s" is not contactable. This could indicate that the service is down or the url is invalid!' % (url))
+    return data
+
+def parse_response(response):
+    data = json.loads(response)
+    if data['status'] == 'ok':
+        nagios_status = nagios.OK
+    elif data['status'] == 'warn':
+        nagios_status = nagios.WARN
+    elif data['status'] == 'crit':
+        nagios_status = nagios.CRIT
+    else:
+        nagios_status = nagios.UNKNOWN
```

Comment:
The existing code is perfectly fine, just sharing a Python idiom with a dict in case it comes in hand some day:

```python
def parse_response_alt(response):
    status = {
      "ok": nagios.OK,
      "warn": nagios.WARN,
      "crit": nagios.CRIT
    }.get(data["status"], nagios.UNKOWN)
    return status, data.get("summary", ""), data.get("details", "")
```

Using `dict.get` vs. `dict[key]` has a subtle difference in that `parse_response` may raise a KeyError for bad input, while `parse_response_alt` doesn't, probably not relevant here.